File: backend/src/main/java/br/com/asa/services/SubdepartmentService.java
package br.com.asa.services;

import br.com.asa.dtos.SubdepartmentDto;
import org.springframework.data.domain.Page;
import java.util.List;

public interface SubdepartmentService {
	Page<SubdepartmentDto> findAll(Integer pageNo, Integer pageSize, String sortBy);
	List<SubdepartmentDto> getBySubdepartment(String search);
	void save(SubdepartmentDto subdepartmentDto);
}

File: frontend/src/services/SubdepartmentService.js
import http from '@/services/http-common'

const path = '/subdepartment'

class SubdepartmentService {
  getAll(currentPage) {
    return http.get(`${path}?pageNo=${currentPage}`)
  }

  getBySubdepartment(search) {
    return http.get(`${path}/search?subdepartment=${search}`)
  }

  new(data) {
    return http.post(`${path}/new`, data)
  }

  edit(data) {
    return http.put(`${path}/edit`, data)
  }
}

export default new SubdepartmentService()
